tools/windows/DatadogAgentInstaller/CustomActions/NpmCustomAction.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using Datadog.CustomActions.Extensions;
+using Microsoft.Deployment.WindowsInstaller;
+using Microsoft.Win32;
+
+namespace Datadog.CustomActions;
+
+public class NpmCustomAction
+{
+    private static ActionResult EnsureNpmServiceDepdendency(ISession session)
+    {
+        try
+        {
+            var addLocal = session.Property("ADDLOCAL");
+            session.Log($"ADDLOCAL={addLocal}");
+            using var systemProbeDef = Registry.LocalMachine.OpenSubKey(@"SYSTEM\CurrentControlSet\Services\datadog-system-probe", true);
+            if (systemProbeDef != null)
+            {
+                if (string.IsNullOrEmpty(addLocal))
+                {
+                    systemProbeDef.SetValue("DependOnService", new[]
+                    {
+                        "datadogagent"
+                    }, RegistryValueKind.MultiString);
+
+                }
+                else if (addLocal.Contains("NPM"))
```

Comment:
the new closed source PR also checks for "ALL", and case insensitively

https://github.com/DataDog/datadog-agent/pull/15263/files#diff-77c57fb5ea05b8089758c23f938c21e4dc67ca9d3692dfbcd2651466c151c20eR196-R198